modules/data-entry/src/main/java/io/uhndata/cards/internal/QuestionMatrixEditor.java

```diff
@@ -0,0 +1,259 @@
+            // if Answer Option is changed
+            if (isAnswerOption(childNode)) {
+                for (String childNodeName : this.currentNodeBuilder.getChildNodeNames()) {
+                    NodeBuilder qNode = this.currentNodeBuilder.getChildNode(childNodeName);
+                    // Remove and copy changed Answer option to every question node
+                    if (isQuestion(qNode)) {
+                        qNode.getChildNode(name).remove();
+                        copyOptionNode(name, childNode, qNode);
+                    }
+                }
+            }
+            // If this is already a QuestionMatrix, there's no need to descend further down
+            return null;
+        }
+
+        return new QuestionMatrixEditor(this.currentNodeBuilder.getChildNode(name));
+    }
+
+    @Override
+    public Editor childNodeDeleted(String name, NodeState before)
+    {
+        if (this.isQuestionMatrixNode) {
+            if (isAnswerOption(before)) {
+                // delete this option from every question node
+                for (String childNodeName : this.currentNodeBuilder.getChildNodeNames()) {
+                    NodeBuilder qNode = this.currentNodeBuilder.getChildNode(childNodeName);
+                    if (isQuestion(qNode)) {
+                        qNode.getChildNode(name).remove();
+                    }
+                }
+            }
+            // If this is already a QuestionMatrix, there's no need to descend further down
+            return null;
+        }
+
+        return new QuestionMatrixEditor(this.currentNodeBuilder.getChildNode(name));
+
+    }
+
+    private void copyOptionNode(final String name, final NodeBuilder source, final NodeBuilder destination)
+    {
+        NodeBuilder newNode = destination.setChildNode(name);
+        String type = source.getProperty("jcr:primaryType").getValue(Type.NAME);
+        newNode.setProperty("jcr:primaryType", type, Type.NAME);
+        String sType = source.getProperty("sling:resourceSuperType").getValue(Type.STRING);
+        newNode.setProperty("sling:resourceSuperType", sType, Type.STRING);
+        if (source.hasProperty("label")) {
+            String label = source.getProperty("label").getValue(Type.STRING);
+            newNode.setProperty("label", label, Type.STRING);
+        }
+        if (source.hasProperty("value")) {
+            String label = source.getProperty("value").getValue(Type.STRING);
+            newNode.setProperty("value", label, Type.STRING);
+        }
+        if (source.hasProperty("defaultOrder")) {
+            String label = source.getProperty("defaultOrder").getValue(Type.STRING);
+            newNode.setProperty("defaultOrder", label, Type.STRING);
+        }
+        if (source.hasProperty("notApplicable")) {
+            Boolean notApplicable = source.getProperty("notApplicable").getValue(Type.BOOLEAN);
+            newNode.setProperty("notApplicable", notApplicable, Type.BOOLEAN);
+        }
+    }
```

Comment:
```suggestion
        if (source.hasProperty("noneOfTheAbove")) {
            Boolean noneOfTheAbove = source.getProperty("noneOfTheAbove").getValue(Type.BOOLEAN);
            newNode.setProperty("noneOfTheAbove", noneOfTheAbove, Type.BOOLEAN);
        }
    }
```
Should this be copied over like the `notApplicable` property? `noneOfTheAbove` options seem to work fine without this property being copied over, but I'm not sure if this being missed would cause issues in the future.

Reply:
Yes, good catch

Reply:
Actually, all works because when displaying forms with question matrices we read all the answer options info from the section and ignore anything specified in the question. Copying the options to the question itself is for filtering purposes only. So technically both the `notApplicable` property and the `noneOfTheAbove` property are unnecessary, but if we're going to copy one we'd better do the same for the other for consistency.

Reply:
Done in 5a652b9.